java-frontend/src/main/java/org/sonar/java/Measurer.java

```diff
@@ -60,6 +62,15 @@ public Measurer(SensorContext context, NoSonarFilter noSonarFilter) {
     public void scanFile(JavaFileScannerContext context) {
       sonarFile = context.getInputFile();
       createCommentLineVisitorAndFindNoSonar(context);
+      if (isSonarQubeSupportingTestNLOC()) {
+        // SonarQube 8.5 supports NLOC for test code, and will take care of computing the difference.
+        saveMetricOnFile(CoreMetrics.NCLOC, new LinesOfCodeVisitor().linesOfCode(context.getTree()));
```

Comment:
These lines can be tested by forcing the context with an inexisting version in the unit Test, for instance by setting the context runtime in given unit test of the measurer:
```
context.setRuntime(
  SonarRuntimeImpl.forSonarQube(org.sonar.api.utils.Version.create(8,5), 
  SonarQubeSide.SCANNER, 
  SonarEdition.COMMUNITY);
```